core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -570,8 +570,11 @@ object Doc {
   /**
    * Invariant: generate is referentially transparent
    */
-  private[paiges] case class LazyDoc(private val generate: () => Doc) extends Doc {
+  private[paiges] class LazyDoc(private val generate: () => Doc) extends Doc {
     lazy val evaluated: Doc = generate()
+    override def canEqual(that: Any): Boolean = false
```

Comment:
I think you need to return `that.isInstanceOf[LazyDoc]`

Reply:
Why is that?  I would think that Scala functions are never considered equal, even if they have the same code.

Reply:
You need reflexivity. `a == a` should be true.